partiql-conformance-tests/build.rs

```diff
@@ -0,0 +1,296 @@
+use ion_rs::value::reader::{element_reader, ElementReader};
+use ion_rs::value::{Element, Sequence, Struct, SymbolToken};
+use ion_rs::IonType;
+use std::ffi::OsStr;
+use std::fs;
+use std::fs::File;
+use std::io::Write;
+use std::path::{Path, PathBuf};
+use walkdir::WalkDir;
+
+const PASS_PARSE_TESTS_DIR: &str = "partiql-tests/partiql-test-data/pass/parser";
+const FAIL_PARSE_TESTS_DIR: &str = "partiql-tests/partiql-test-data/fail/parser";
+const PARSER_IMPORT: (&str, &str) = ("partiql_parser", "lalr_parse");
+
+fn main() {
+    let pass_parse_test_files: Vec<PathBuf> = all_ion_files_in(PASS_PARSE_TESTS_DIR);
+    let fail_parse_test_files: Vec<PathBuf> = all_ion_files_in(FAIL_PARSE_TESTS_DIR);
+
+    for pass_parse_test_file in &pass_parse_test_files {
+        let mut scope = create_pass_parse_scope(pass_parse_test_file);
+        scope.import(PARSER_IMPORT.0, PARSER_IMPORT.1);
+        let full_file_name = to_full_file_name("pass_parse", pass_parse_test_file);
+        let dest_path: PathBuf = ["tests", full_file_name.as_str()].iter().collect();
+        File::create(dest_path)
+            .expect("File creation failed")
+            .write_all(scope_to_formatted_string(scope).as_bytes())
+            .unwrap_or_else(|error| panic!("Failure when writing to file: {:?}", error));
+    }
+
+    for fail_parse_test_file in &fail_parse_test_files {
+        let mut scope = create_fail_parse_scope(fail_parse_test_file);
+        scope.import(PARSER_IMPORT.0, PARSER_IMPORT.1);
+        let full_file_name = to_full_file_name("fail_parse", fail_parse_test_file);
+        let dest_path: PathBuf = ["tests", full_file_name.as_str()].iter().collect();
+        File::create(dest_path)
+            .expect("File creation failed")
+            .write_all(scope_to_formatted_string(scope).as_bytes())
+            .unwrap_or_else(|error| panic!("Failure when writing to file: {:?}", error));
+    }
+}
+
+/// Returns a vector of all .ion files in the directory `dir`
+fn all_ion_files_in(dir: &str) -> Vec<PathBuf> {
+    let ion_file_extension: &OsStr = OsStr::new("ion");
+    let ion_file_iterator = WalkDir::new(dir)
+        .into_iter()
+        .map(|entry| {
+            entry.unwrap_or_else(|error| panic!("Failure during dir traversal: {:?}", error))
+        })
+        .filter(|entry| entry.path().is_file())
+        .map(|entry| entry.path().to_owned())
+        .filter(|f| f.extension() == Some(ion_file_extension));
+    ion_file_iterator.collect()
+}
+
+/// Returns a file name with prefix prepended and 'rs' as the extension
+fn to_full_file_name(prefix: &str, ion_file: &Path) -> String {
+    format!(
+        "{}_{}",
+        prefix,
+        ion_file
+            .with_extension("rs")
+            .file_name()
+            .unwrap()
+            .to_str()
+            .unwrap()
+    )
+}
+
+/// Converts the scope to a String that is passed through codegen's default formatter
+fn scope_to_formatted_string(scope: Scope) -> String {
```

Comment:
I think this is [Scope.to_string](https://docs.rs/codegen/0.1.3/codegen/struct.Scope.html#method.to_string)